enterprise/cli/proxyserver.go

```diff
@@ -77,10 +77,15 @@ func (*RootCmd) proxyServer() *clibase.Cmd {
 			Flag:        "primary-access-url",
 			Env:         "CODER_PRIMARY_ACCESS_URL",
 			YAML:        "primaryAccessURL",
-			Default:     "",
-			Value:       &primaryAccessURL,
-			Group:       &externalProxyOptionGroup,
-			Hidden:      false,
+			Required:    true,
+			Value: clibase.Validate(&primaryAccessURL, func(value *clibase.URL) error {
+				if !(value.Scheme == "http" || value.Scheme == "https") {
+					return xerrors.Errorf("'--primary-access-url' value must be http or https: url=%s", primaryAccessURL.String())
```

Comment:
Ideally instead of showing the flag (`--primary-access-url`) we should the correct value source. That should probably get populated within clibase since the code would be duplicated in every validator.

Reply:
This error message is bad, the `Validate` handling code should return an error that has the flag name and the `ValueSource` in it. This error message should just say `value must be http or https: url=%s`